### PR TITLE
Add floatLE doubleLE uInt32LE and int32LE read+write methods

### PR DESCRIPTION
**Pull request by HDegroote**
# Add floatLE, doubleLE uInt32LE and int32LE read and write methods.

Please review with some care, as I don't have much experience with working directly with buffers. 

The pull-request is feature-complete, but marked as a draft because I'm not sure which style is preferred for the tests (I've never used brittle before). See for example test/node.mjs: writeDoubleLE defines multiple tests within the same block, whereas readDoubleLE uses subtests to do the same--which is preferred? 

Note: I also added a test to check that index.js and browser.js export the exact same functions. If this is not relevant, I'll remove it again.

### CODE REVIEW

**Comment by HDegroote** [comment]
Note: I didn't add anything to the benchmarks

**Comment by mafintosh** [comment]
@kasperisager any issues with this?

File: browser.js
Comment (kasperisager) [inline browser.js:399]: Logical nullish assignment is only supported from Node.js 15 and onwards so let's do `if (offset === undefined) offset = 0` as in the existing functions instead.

File: browser.js
Comment (kasperisager) [inline browser.js:405]:
```suggestion
  return offset + 8
```

File: browser.js
Comment (kasperisager) [inline browser.js:401]:
```suggestion
  const view = new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength)
```

A typed array is a view over an underlying buffer, possibly only containing a subset of it. The data view must reflect this so that `offset` remains relative to `buffer.byteOffset`.

**Review by kasperisager — CHANGES_REQUESTED** [review]
Thanks for the PR! 👏 I've left some comments for `writeDoubleLE()` that apply to the remaining functions as well.

**Comment by HDegroote** [comment]
Thanks for the comments! I think I took them all into account. I'm sorry about missing that sub-buffer case, I added tests for it.

I also would like to double-check something about the tests, because I'm a bit confused about how the standard test-runner reports errors. 
With how I structured the tests at the moment, when a nested test fails, this failure doesn't seem to 'propagate' to the enclosing test. So when using the standard runner (tap), it reports a failure for that specific subtest, but the parent test and the overall run show up as ok. When I run the tests with the 'spec' runner, I do get a summary at the end with the failed tests, so it's obvious when they fail. 
Regardless of which test runner is used, the exit code is 1 in case of failure, so it's not that tap swallows the errors, it just reports them in a way where it's easy to miss one. 
I simply wanted to make sure that the reporting behaviour I described is as expected, or if I should restructure the tests somehow.

**Comment by kasperisager** [comment]
> I'm sorry about missing that sub-buffer case, I added tests for it.

No need to apologise, it's an easy thing to miss! Thanks so much for the test as well!

> I also would like to double-check something about the tests, because I'm a bit confused about how the standard test-runner reports errors.

That is very likely just a quirk of how our test library outputs TAP. @mafintosh is currently reworking it and has simplified the subtest flow quite a bit; it ought to report errors as expected.

**Review by kasperisager — APPROVED** [review]

**Comment by kasperisager** [comment]
Let me know if there's anything else you'd like to add before we merge. If not, feel free to mark as ready for review and then I'll merge when CI is green.

Thanks again!

**Comment by HDegroote** [comment]
I updated some browser tests to also use the hierarchical approach, to make everything consistent. I think it's good to go now

**Pull request merged**
